pkg/controller/workload/job/job_controller.go

```diff
@@ -170,11 +183,27 @@ func (r *JobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 
 	// 3. handle a finished job
 	if jobFinished {
-		added := false
-		wl.Status.Conditions, added = appendFinishedConditionIfNotExists(wl.Status.Conditions, jobFinishedCond)
-		if !added {
+		if workload.InCondition(wl, kueue.WorkloadFinished) {
+			return ctrl.Result{}, nil
+		}
+		condition := generateFinishedCondition(jobFinishedCond)
+		apimeta.SetStatusCondition(&wl.Status.Conditions, condition)
+		err := r.client.Status().Update(ctx, wl)
+		if err != nil {
+			log.Error(err, "Updating workload status")
+		}
+		return ctrl.Result{}, err
+	}
+
+	// handle a job when waitForPodsReady is enabled
+	if r.waitForPodsReady {
```

Comment:
you are always returning if you enter this branch. We should continue doing the other checks on the Job

Reply:
Fixed